apps/studio/src/features/editing-experience/components/MoveResourceModal/MoveResourceModal.tsx

```diff
@@ -0,0 +1,215 @@
+    // the last item here represents his current view
+    // NOTE: We should always start the stack from `/` (root)
+    // so that the user will see a full overview of their site structure
+    const [resourceStack, setResourceStack] = useState<MovedItem[]>([])
+    const setMovedItem = useSetAtom(moveResourceAtom)
+    const [{ title }] = trpc.resource.getMetadataById.useSuspenseQuery({
+      resourceId,
+    })
+    const curResourceId = resourceStack[resourceStack.length - 1]?.resourceId
+    const [children] = trpc.resource.getChildrenOf.useSuspenseQuery({
+      resourceId: curResourceId ?? null,
+    })
+    const utils = trpc.useUtils()
+    const toast = useToast({ status: "success" })
+    const { mutate, isLoading } = trpc.resource.move.useMutation({
+      onError: (err) => {
+        toast({
+          title: "Failed to move resource",
+          status: "error",
+          description: err.message,
+        })
+      },
+      onSettled: () => {
+        // TODO: actually close the modal
+        setMovedItem(null)
+      },
+      onSuccess: () => {
+        void utils.page.readPageAndBlob.invalidate()
+        void utils.page.list.invalidate({
+          // TODO: Update backend `list` to use the proper schema
+          resourceId: movedItem?.resourceId
+            ? Number(movedItem.resourceId)
+            : undefined,
+        })
+        // NOTE: We might want to have smarter logic here
+        // and invalidate the new + old folders
+        void utils.folder.readFolder.invalidate()
+        toast({ title: "Resource moved!" })
+      },
+    })
+
+    const movedItem = useAtomValue(moveResourceAtom)
+    // TODO: need to update move dest
+    const moveDest = resourceStack[resourceStack.length - 1]
+    const onBack = () => {
+      setResourceStack((prev) => prev.slice(0, -1))
+    }
+
+    return (
+      <ModalContent>
+        <ModalHeader>Move "{title}" to...</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <VStack alignItems="flex-start" spacing="1.25rem">
+            <Infobox size="sm" w="full">
+              Moving a page or folder changes its URL, effective immediately
+            </Infobox>
+            <Box
+              borderRadius="md"
+              border="1px solid"
+              borderColor="base.divider.strong"
+              w="full"
+              py="0.75rem"
+              px="0.5rem"
+            >
+              {resourceStack.length > 0 ? (
+                <Button
+                  variant="clear"
+                  w="full"
+                  justifyContent="flex-start"
+                  color="base.content.default"
+                  onClick={onBack}
+                >
+                  <HStack spacing="0.25rem" color="interaction.links.default">
+                    <BiLeftArrowAlt />
+                    <Text textStyle="caption-1">Back to parent folder</Text>
+                  </HStack>
+                </Button>
```

Comment:
supernit: in figma, seems to be using a `Link` component. We can utilise the `as` prop to reduce restyling efforts (and prevent drift):

```suggestion
                <Link
                  size="xs"
                  as="button" // this allows usage of Link styles, but render as button
                  onClick={onBack}
                >
                  <HStack spacing="0.25rem">
                    <BiLeftArrowAlt />
                    <Text>Back to parent folder</Text>
                  </HStack>
                </Button>
```

Reply:
donezxc